relay-server/src/utils/rate_limits.rs

```diff
@@ -142,6 +142,84 @@ fn infer_event_category(item: &Item) -> Option<DataCategory> {
     }
 }
 
+/// Quantity metrics for a single category of attachments.
+///
+/// Tracks both the count of attachments and size in bytes.
+#[derive(Clone, Copy, Debug, Default)]
+pub struct AttachmentQuantities {
+    pub count: usize,
+    pub bytes: usize,
+}
+
+impl AttachmentQuantities {
+    pub fn is_empty(&self) -> bool {
+        self.count == 0 && self.bytes == 0
+    }
+}
+
+/// Aggregated attachment quantities grouped by [`ParentType`].
+///
+/// This separation is necessary since rate limiting logic varies by [`ParentType`].
+#[derive(Clone, Copy, Debug, Default)]
+pub struct AttachmentsQuantities {
+    /// Quantities of Event Attachments.
+    ///
+    /// See also: [`AttachmentParentType::Event`].
+    event: AttachmentQuantities,
+    /// Quantities of trace V2 Attachments.
+    trace: AttachmentQuantities,
+    /// Quantities of span V2 Attachments.
+    span: AttachmentQuantities,
+}
+
+impl AttachmentsQuantities {
+    /// Returns the total count of all attachments across all parent types.
+    pub fn count(&self) -> usize {
+        let AttachmentsQuantities {
+            event:
+                AttachmentQuantities {
+                    count: event_count,
+                    bytes: _,
+                },
+            trace:
+                AttachmentQuantities {
+                    count: trace_count,
+                    bytes: _,
+                },
+            span:
+                AttachmentQuantities {
+                    count: span_count,
+                    bytes: _,
+                },
+        } = self;
+
+        event_count + trace_count + span_count
```

Comment:
nit: IMO it's OK to not destructure everything here, since you're only interested in the `count` field:

```suggestion
        let AttachmentsQuantities {event, trace, span} = self;
        event.count + trace.count + span.count
```